From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thibaut Sautereau <clipos@ssi.gouv.fr>
Date: Wed, 25 Jul 2018 14:09:34 +0200
Subject: [PATCH 09/32] clip: vserver/grsec: Fix lockref access operations

grsecurity uses atomic helpers to handle lockrefs, so patch VServer
hunks to use that too.

Patch: 1508_vserver_grsec_lockref-fixes.patch
---
 fs/namei.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/namei.c b/fs/namei.c
index 8e2d551ebbbfa..508d5f3699339 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -5130,13 +5130,13 @@ void dump_path(const char *name, struct path *path)
 			  path->mnt->mnt_sb ? path->mnt->mnt_sb->s_count : -1,
 			  path->mnt->mnt_sb ? atomic_read(&path->mnt->mnt_sb->s_active) : -1,
 			  path->mnt->mnt_root,
-			  path->mnt->mnt_root ? path->mnt->mnt_root->d_lockref.count : -1);
+			  path->mnt->mnt_root ? __lockref_read(&path->mnt->mnt_root->d_lockref) : -1);
 
 	if (path && path->dentry)
 		vxdprintk(VXD_CBIT(misc, 3),
 			  "%s: path dentry=%p[#%d]", name,
 			  path->dentry,
-			  path->dentry ? path->dentry->d_lockref.count : -1);
+			  path->dentry ? __lockref_read(&path->dentry->d_lockref) : -1);
 }
 
 static inline
-- 
2.17.0

